bundles/org.openhab.binding.dali/src/main/java/org/openhab/binding/dali/internal/handler/DaliRgbHandler.java

```diff
@@ -0,0 +1,130 @@
+
+    static final int Switch100Percent = 254;
+
+    public DaliRgbHandler(Thing thing) {
+        super(thing);
+    }
+
+    @Override
+    public void initialize() {
+        initDeviceState();
+    }
+
+    @Override
+    public void bridgeStatusChanged(ThingStatusInfo bridgeStatusInfo) {
+        if (bridgeStatusInfo.getStatus() == ThingStatus.ONLINE
+                && getThing().getStatusInfo().getStatusDetail() == ThingStatusDetail.BRIDGE_OFFLINE) {
+            initDeviceState();
+        } else if (bridgeStatusInfo.getStatus() == ThingStatus.OFFLINE) {
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.BRIDGE_OFFLINE);
+        }
+    }
+
+    public void initDeviceState() {
+        Bridge bridge = getBridge();
+
+        if (bridge == null) {
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.CONFIGURATION_ERROR, "No bridge configured");
+        } else if (bridge.getStatus() == ThingStatus.ONLINE) {
+            updateStatus(ThingStatus.ONLINE);
+        } else {
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.BRIDGE_OFFLINE);
+        }
+    }
+
+    @Override
+    public void handleCommand(ChannelUID channelUID, Command command) {
+        try {
+            if (CHANNEL_COLOR.equals(channelUID.getId())) {
+                List<Integer> outputs = List.of(
+                        ((BigDecimal) this.thing.getConfiguration().get(TARGET_ID_R)).intValueExact(),
+                        ((BigDecimal) this.thing.getConfiguration().get(TARGET_ID_G)).intValueExact(),
+                        ((BigDecimal) this.thing.getConfiguration().get(TARGET_ID_B)).intValueExact());
+
+                if (command instanceof HSBType) {
+                    PercentType[] rgb = ((HSBType) command).toRGB();
+
+                    for (int i = 0; i < 3; i++) {
+                        byte dimmValue = (byte) ((rgb[i].floatValue() * Switch100Percent) / 100);
+                        getBridgeHandler()
+                                .sendCommand(new DaliDAPCCommand(DaliAddress.Short(outputs.get(i)), dimmValue));
+                    }
+                } else if (command instanceof OnOffType) {
+                    if ((OnOffType) command == OnOffType.ON) {
+                        for (Integer output : outputs) {
+                            getBridgeHandler().sendCommand(
+                                    new DaliDAPCCommand(DaliAddress.Short(output), (byte) Switch100Percent));
+                        }
+                    } else {
+                        for (Integer output : outputs) {
+                            getBridgeHandler().sendCommand(DaliStandardCommand.Off(DaliAddress.Short(output)));
+                        }
+                    }
+                }
+            }
+        } catch (DaliException e) {
+            logger.warn("Error handling command: {}", e.getMessage());
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.COMMUNICATION_ERROR, e.getMessage());
```

Comment:
Can you do a FIND on 'updateStatus(' and then check if you are doing a logger at the same time and change all of them please?